src/main/kotlin/registration/EthFreeWalletsProvider.kt

```diff
@@ -16,29 +15,33 @@ import sidechain.iroha.util.getRelays
 // TODO Prevent double relay accounts usage (in perfect world it is on Iroha side with custom code). In real world
 // on provider side with some synchronization.
 class EthFreeWalletsProvider(
-    val irohaConfig: IrohaConfig,
-    val keypair: Keypair,
-    val notaryIrohaAccount: String,
-    val registrationIrohaAccount: String
+    private val irohaConfig: IrohaConfig,
+    private val keypair: Keypair,
+    private val notaryIrohaAccount: String,
+    private val registrationIrohaAccount: String
 ) {
-    val irohaNetwork = IrohaNetworkImpl(irohaConfig.hostname, irohaConfig.port)
+    private val irohaNetwork = IrohaNetworkImpl(irohaConfig.hostname, irohaConfig.port)
     /**
-     * Get free ethereum relay wallet.
+     * Get first free ethereum relay wallet.
      * @return free ethereum relay wallet
      */
     fun getWallet(): Result<String, Exception> {
+        return getWallets().map { freeWallets -> freeWallets.first() }
+    }
+
+    /**
+     * Get all free ethereum relay wallets
+     * @return free ethereum relay wallets
+     */
+    fun getWallets(): Result<Set<String>, Exception> {
```

Comment:
may be rename to getFreeRelays()?

Reply:
what about renaming the whole class as well?

Reply:
EthFreeRelayProvider